File: workspace/pset2/caesar/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <stdlib.h>

int main(int argc, string argv[])
{
    string plainText;
    
    if (argc != 2)
    {
        printf ("No command line argument passed \n");
        return 1; 
    }
    
    int key = atoi(argv[1]);
    printf("plaintext: ");
    plainText = GetString();
    
    for(int i = 0, n = strlen(plainText); i < n; i++)
    {
        
        if (plainText[i] >= 'A' && plainText[i] <= 'Z')
        {
            plainText[i] = (((plainText[i] - 65) + key) % 26) + 65;
        }
        else if (plainText[i] >= 'a' && plainText[i] <= 'z')
        {
            
            plainText[i] = (((plainText[i] - 97) + key) % 26) + 97;
        }
    }
    
    printf("ciphertext: %s",plainText);
    printf("\n");
    return 0;
}

File: Pset6_Shashank/caesar.py
import cs50
import sys

def main():
    if len(sys.argv) != 2:
        print("No command line argument passed")
        exit(1)
    
    key = int(sys.argv[1])
    t = []
    print("plaintext: ",end='')
    plaintext = cs50.get_string()
    print("ciphertext: ",end='')
    for symbol in plaintext:
        if symbol.isalpha():
            t.append(caesar(symbol, key))
        else:
            t.append(symbol)
                
    print("".join(t))
    exit(0)

def caesar(char, key):
    if char.isupper():
        return chr(((ord(char) - 65 + key) % 26) + 65)
    else:
        return chr(((ord(char) - 97 + key) % 26) + 97)
        
if __name__ == "__main__":
    main()